### PR TITLE
Nitrpg: introduce game events

### PR DESCRIPTION
**Pull request by Morriar**
Events allow us to keep track of the history.

### CODE REVIEW

**Comment by R4PaSs** [comment]
+1 !

File: contrib/nitrpg/src/events.nit [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+	# GameEvent uniq id used for storage.
+	var internal_id: String is noinit
+
+	# Date and time of the event.
+	var time: ISODate is noinit, writable
+
+	# An event initialized at now `time`.
+	init do
+		internal_id = "{get_time}{object_id}{100.rand}"
```
Comment (privat) [inline contrib/nitrpg/src/events.nit]: 100.rand?

Comment (Morriar) [inline contrib/nitrpg/src/events.nit]: To avoid collisions on keys. A better idea/ID ?

File: contrib/nitrpg/src/templates/templates_events.nit [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+	# See `TplEvent`
+	fun tpl_event: TplEvent do
+		if kind == "pull_open" then
+			return new TplPullOpened(self)
+		else if kind == "pull_merged" then
+			return new TplPullMerged(self)
+		else if kind == "pull_review" then
+			return new TplPullReview(self)
+		end
+		abort
```
Comment (privat) [inline contrib/nitrpg/src/templates/templates_events.nit]: bigre! ça plaisante pas!

File: contrib/nitrpg/src/events.nit (hunk 1 above)
Comment (privat) [inline contrib/nitrpg/src/events.nit]: time and object_id seems enough for unicity. is the rand also for security, a hash of the thing can do the trick.

**Comment by privat** [comment]
+1 this is good enough for me.

**Pull request merged**